cedar-policy-validator/src/custom_schema/err.rs

```diff
@@ -0,0 +1,322 @@
+use std::{
+    collections::HashMap,
+    fmt::Display,
+    ops::{Deref, DerefMut},
+};
+
+use cedar_policy_core::parser::{err::expected_to_string, unescape::UnescapeError, Loc, Node};
+use lalrpop_util as lalr;
+use lazy_static::lazy_static;
+use miette::{Diagnostic, LabeledSpan, SourceSpan};
+use nonempty::NonEmpty;
+use smol_str::SmolStr;
+use thiserror::Error;
+
+use super::ast::Str;
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub enum UserError {
+    EmptyList,
+    StringEscape(NonEmpty<UnescapeError>),
+}
```

Comment:
if this is an error type, should it derive `Error` according to our style in Core/Validator crates?